docs/messaging/messaging-metrics.md

```diff
@@ -0,0 +1,194 @@
+
+| Value  | Description |
+|---|---|
+| `_OTHER` | A fallback error value to be used when the instrumentation doesn't define a custom value. |
+
+`network.transport` has the following list of well-known values. If one of them applies, then the respective value MUST be used, otherwise a custom value MAY be used.
+
+| Value  | Description |
+|---|---|
+| `tcp` | TCP |
+| `udp` | UDP |
+| `pipe` | Named or anonymous pipe. |
+| `unix` | Unix domain socket |
+
+`network.type` has the following list of well-known values. If one of them applies, then the respective value MUST be used, otherwise a custom value MAY be used.
+
+| Value  | Description |
+|---|---|
+| `ipv4` | IPv4 |
+| `ipv6` | IPv6 |
+<!-- endsemconv -->
+
+## Producer metrics
+
+### Metric: `messaging.publish.duration`
+
+This metric is [required][MetricRequired].
+
+When this metric is reported alongside a messaging publish span, the metric value SHOULD be the same as the corresponding span duration.
+
+This metric SHOULD be specified with
+[`ExplicitBucketBoundaries`](https://github.com/open-telemetry/opentelemetry-specification/tree/v1.26.0/specification/metrics/api.md#instrument-advice)
+of `[ 0.005, 0.01, 0.025, 0.05, 0.075, 0.1, 0.25, 0.5, 0.75, 1, 2.5, 5, 7.5, 10 ]`.
+
+<!-- semconv metric.messaging.publish.duration(metric_table) -->
+| Name     | Instrument Type | Unit (UCUM) | Description    |
+| -------- | --------------- | ----------- | -------------- |
+| `messaging.publish.duration` | Histogram | `s` | Measures the duration of publish operation. |
+<!-- endsemconv -->
+
+### Metric: `messaging.publish.messages`
+
+This metric is [required][MetricRequired] when messaging system supports batch publishing. It's [opt-in][MetricOptIn] when messaging system does not support batch publishing since message count can be derived from `messaging.publish.duration` histogram.
+
+_Note: The need to report `messaging.publish.messages` depends on the messaging system capabilities and not application scenarios or client library limitations.
+For example, RabbitMQ does not support batch publishing and corresponding instrumentations don't need to report `messaging.publish.messages`. Kafka supports both, single and batch publishing, and instrumentations MUST report `messaging.publish.messages` counter regardless of application scenarios or APIs available in the client library.
+
+<!-- semconv metric.messaging.publish.messages(metric_table) -->
+| Name     | Instrument Type | Unit (UCUM) | Description    |
+| -------- | --------------- | ----------- | -------------- |
+| `messaging.publish.messages` | Counter | `{message}` | Measures the number of published messages. |
+<!-- endsemconv -->
+
+## Consumer metrics
+
+### Metric: `messaging.receive.duration`
+
+This metric is [required][MetricRequired] when messaging system supports poll-based receive operations.
```

Comment:
```suggestion
This metric is [required][MetricRequired] for poll-based receive operations.
```

A nit, but it almost sounds as if this metric is required if "receive" operations are supported by system, regardless of whether the users uses it.